src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -2510,6 +2510,11 @@ std::pair<std::shared_ptr<Pane>, std::shared_ptr<Pane>> Pane::_Split(SplitDirect
     // Clear out our ID, only leaves should have IDs
     _id = {};
 
+    // Make sure to propagate our _isDefTermSession value
+    WalkTree([&](auto p) {
+        p->_isDefTermSession = _isDefTermSession;

```

Comment:
Also, don't we need to bring the value back when you close a split and the Parent becomes a Leaf again?

And like, we need to correctly handle the state when you open a DefTerm, split it, and close the DefTerm. The remaining leaf should NOT have this set to true ..

Reply:
> Also, don't we need to bring the value back when you close a split and the Parent becomes a Leaf again?

The parent's `_isDefTermSession` doesn't get overwritten when a split happens! So we don't need to bring the value back, it is already there

> And like, we need to correctly handle the state when you open a DefTerm, split it, and close the DefTerm. The remaining leaf should NOT have this set to true ..

Ah, this was my misunderstanding, I thought any splits made while in a def term session should also be def term, can fix that

Reply:
> Also, don't we need to bring the value back when you close a split and the Parent becomes a Leaf again?

Oh I think I understand what you mean now....

Pane A is def term, and `_Split` is called -
- Pane A becomes a parent, Pane B inherits the control/profile from Pane A, and Pane C is the new child
- Pane B should be considered a def term session, Pane C should not

So, when `Close` is called on Pane B, A inherits the control/profile from C and should also inherit C's `isDefTermSession`

In other words, the `isDefTermSession` value should be migrated back and forth between children and parents whenever the control/profile is migrated

Reply:
So here's how I think about it. A def term session is *a session that originated outside Terminal.* That is the defining characteristic -- it isn't something that can be inherited by something that comes from inside Terminal, because that would violate that invariant.

Reply:
Fixed :)